### ISSUE TITLE
Regression: single Polymorphic field causes descriptor resolution failure

### ISSUE THREAD

**Issue by altavir** [pdvrieze/xmlutil#18]
I have an element which could contain either element with type `A` or with type `B` distinguished by its tag name. In previous versions I could write something like
```kotlin
    @XmlPolyChildren(arrayOf("physvol","divisionvol"))
    @Polymorphic
    var placement: GDMLPlacement? = null
```
and it worked. Now if I do that, then I get Stackoverflow exception caused by the fact that descriptor could not be located (it is fixed in 0.13.0, but latest version is not compatible with xmlutil). It is very hard to understand what exactly causes the error.

Current workaround is to replace single element by a list:
```kotlin
    var placement = ArrayList<@Polymorphic GDMLPlacement>()
```

**Comment by pdvrieze** [comment]
There is a bug in the older version of kotlinx.serialization where the code that creates an exception when a serialdescriptor does not contain a field will try to look up the serial descriptor to give a nicer error message (which then throws the same exception again). I will update the library to use newer kotlinx.serialization where I actually contributed the fix.

**Comment by altavir** [comment]
I do understand that. The problem is not with reporting an exception, but with exception being raised in the first place. The code should work.

**Comment by pdvrieze** [comment]
The exception is due to the fact that not all descriptors properly support getting child serializers (even though they do list an element count). `PolymorphicClassDescriptor` is one of those. There is unfortunately no way for code to know what will happen if it calls getElementDescriptor with a valid index (without hardcoding special cases for specific descriptors/serializers). Therefore the only way to handle this is by catching the exception (I had a hardcoded workaround but clearly that didn't work for your case).

Anyway I just released v0.13.0.0 that works against kotlinx.serialization-0.13.0
